napari/layers/base/base.py

```diff
@@ -581,6 +609,13 @@ def cursor_size(self, cursor_size):
         self.events.cursor_size(cursor_size=cursor_size)
         self._cursor_size = cursor_size
 
+    def set_view_slice(self):
+        if self._is_dask_data:
+            with dask.config.set({"optimization.fuse.active": False}):
+                self._set_view_slice()
+        else:
+            self._set_view_slice()
```

Comment:
@tlambert03 as a reminder from our conversation on Zulip, could you add a link to the dask code that this affects?

Reply:
I also wonder if we should just do the config set regardless of `_is_dask_data`.

Reply:
sure! then I guess we don't need to care if it's a dask array at all. should we still emit a one-time warning?

Reply:
that's a good question! =P Not sure.

Reply:
Another approach would be to create a context manager that is a no-op if there is no dask data within the `self._config_dask()` method proposed above. then here we would always do:

```python
with self.dask_optimized_slicing():
    self._set_view_slice()
```

Reply:
love this, did it. and I think it looks much better now!